From 77963acb3440638eb95a1c86d25dd68212ea9564 Mon Sep 17 00:00:00 2001
From: "P.K. Lee" <p.k.lee@quantatw.com>
Date: Thu, 17 Jul 2025 14:35:52 +0800
Subject: [PATCH] ARM: dts: aspeed: ventura: add missing cable presence gpios

Since the 9 x cable presence GPIOs for 4 x AALC cables, 5 x Rack level
leak cables are missing from the Ventura DTS, we added those to the
device tree.

LORE link:
https://lore.kernel.org/all/20250716094329.1069203-1-pkleequanta@gmail.com/T/#u

Signed-off-by: P.K. Lee <p.k.lee@quantatw.com>
---
 .../aspeed/aspeed-bmc-facebook-ventura.dts    | 51 +++++++++++++++++++
 1 file changed, 51 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
index 54528a560f77..d8bb844065df 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
@@ -1227,6 +1227,57 @@ power-monitor@69 {
 		compatible = "pmbus";
 		reg = <0x69>;
 	};
+
+	gpio@19 {
+		compatible = "nxp,pca9555";
+		reg = <0x19>;
+		gpio-controller;
+		#gpio-cells = <2>;
+
+		gpio-line-names =
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"wIT_GEAR_RPU_2_LINK_PRSNT_SPARE_N_R", "wIT_GEAR_RPU_2_LINK_PRSNT_N_R",
+		"wIT_GEAR_RPU_LINK_PRSNT_SPARE_N_R", "wIT_GEAR_RPU_LINK_PRSNT_N_R",
+		"", "",
+		"", "";
+	};
+
+	gpio@1a {
+		compatible = "nxp,pca9555";
+		reg = <0x1a>;
+		gpio-controller;
+		#gpio-cells = <2>;
+
+		gpio-line-names =
+		"wPRSNT_LEAK1_SENSOR_R_PLD_N", "wPRSNT_LEAK0_SENSOR_R_PLD_N",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "wPRSNT_LEAK2_SENSOR_R_PLD_N";
+	};
+
+	gpio@1b {
+		compatible = "nxp,pca9555";
+		reg = <0x1b>;
+		gpio-controller;
+		#gpio-cells = <2>;
+
+		gpio-line-names =
+		"", "",
+		"", "",
+		"", "wPRSNT_LEAK4_SENSOR_R_PLD_N",
+		"wPRSNT_LEAK3_SENSOR_R_PLD_N", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "";
+	};
 };
 
 &i2c14 {
-- 
2.43.0

